codespell_lib/_codespell.py

```diff
@@ -655,24 +655,37 @@ def parse_options(
     return options, parser, used_cfg_files
 
 
-def parse_ignore_words_option(ignore_words_option: List[str]) -> Set[str]:
+def parse_ignore_words_option(
+    ignore_words_option: List[str],
+) -> Tuple[Set[str], Set[str]]:
     ignore_words: Set[str] = set()
+    ignore_words_cased: Set[str] = set()
     if ignore_words_option:
         for comma_separated_words in ignore_words_option:
-            ignore_words.update(
-                word.strip() for word in comma_separated_words.split(",")
-            )
-    return ignore_words
+            for word in comma_separated_words.split(","):
+                word = word.strip()
+                if word == word.lower():
+                    ignore_words.add(word)
+                else:
+                    ignore_words_cased.add(word)
+    return (ignore_words, ignore_words_cased)
 
 
 def build_exclude_hashes(filename: str, exclude_lines: Set[str]) -> None:
     with open(filename, encoding="utf-8") as f:
         exclude_lines.update(line.rstrip() for line in f)
 
 
-def build_ignore_words(filename: str, ignore_words: Set[str]) -> None:
+def build_ignore_words(
+    filename: str, ignore_words: Set[str], ignore_words_cased: Set[str]
+) -> None:
     with open(filename, encoding="utf-8") as f:
-        ignore_words.update(line.strip() for line in f)
+        for line in f:
+            word = line.strip()
+            if word == word.lower():
+                ignore_words.add(word)
+            else:
+                ignore_words_cased.add(word)
```

Comment:
Perhaps factorize common code in a single function:
```python
def classify_ignore_words(words: Iterable[str], ignore_words: Set[str], ignore_words_cased: Set[str]) -> None:
    for word in words:
        word = words.strip()
        if word == word.lower():
            ignore_words.add(word)
        else:
            ignore_words_cased.add(word)
```
Not sure whether file `f` qualifies as an "iterable of `str`".

Reply:
good idea, I implemented this